### DISCUSSION TITLE
New REST API endpoint to get all prompts

### DISCUSSION THREAD

**Discussion by zoltan-fedor** [langfuse/langfuse#1770] [Ideas]
### Describe the feature or potential improvement

Currently through the API one can only retrieve a single prompt by providing its name (https://api.reference.langfuse.com/#get-/api/public/prompts), but there is no way to list of prompts and then retrieve them by their name.

It would be useful to have a `Get a list of prompts` REST API endpoint, just like there is a `Get a list of observations` one.

This could allow us to create "dynamic prompts" - where just by creating a new prompt in Langfuse the behaviour of the LLM app could be enhanced, instead of currently when the hard coding of the prompt name is required into the LLM app.

For example if you are building with the multi-agent pattern, then these "dynamic prompts" could allow you to create additional agents into the multi-agent architecture just by creating new prompts in Langfuse.

**Comment by marcklingen** [comment]
WIP PR: https://github.com/langfuse/langfuse/pull/1402

**Comment by zoltan-fedor** [comment]
Thanks @marcklingen ,
Hopefully that PR will get merged soon, it seems it is going on for a while.

**Reply by marcklingen** [reply]
trying to get this merged asap, I know it has been wip for quite some time
